app/controlplane/internal/service/organization.go

```diff
@@ -158,3 +169,23 @@ func (s *OrganizationService) UpdateMembership(ctx context.Context, req *pb.Orga
 
 	return &pb.OrganizationServiceUpdateMembershipResponse{Result: bizMembershipToPb(m)}, nil
 }
+
+func (s *OrganizationService) canCreateOrganization(ctx context.Context) (bool, error) {
+	// Restricted org creation is disabled, allow creation
+	if !s.enforcer.RestrictOrgCreation {
+		return true, nil
+	}
+
+	m := entities.CurrentMembership(ctx)
+	for _, rm := range m.Resources {
```

Comment:
Maybe here you can restrict to memberships with `ResourceType == "instance"` just to avoid unnecessary enforcer calls.